### PR TITLE
construction graph pizza

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
title

- made pizza construction graph based
- fixed margherita pizza taking 4 tomatoes not cheese
- added mouse pizza
- added recipe for arnold's pizza

TODO:
- pizza using food layer visualizer thing
- implement pizza recipe sprite copying properly
- update pizza name after adding a topping instead of only after cooking
- change construction graph to support a node with an edge to iself to avoid need for stupid hack
- banana peel topping sprite
- reagent combining gaming
- support adding components to the final meal for arnolds pizza functionality
- support adding reagents for recipes like omnizine/nutriment on top of the base reagent combination
- idk

## Why / Balance
part of pro design doc

## Technical details
ConstructedMealRecipe instead of microwave
FinishMeal step that does the stuff

## Media
- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
uh

**Changelog**
:cl:
- tweak: Pizza is no longer made in the microwave, you roll dough, add sauce and toppings, then bake it!

### CODE REVIEW

**Comment by lzk228** [comment]
god almighty........

**Comment by Doru991** [comment]
The modular food... the modular food is real ‼️

**Comment by lzk228** [comment]
holy shit label required

File: Resources/Prototypes/Recipes/Construction/Graphs/food/pizza.yml [hunk 1]
```diff
@@ -5,10 +5,64 @@
...
     entity: FoodDoughFlat
-  # TODO: add stuff for adding ingredients then cooking the pizza, wsci
+    edges:
+    - to: toppings
+      conditions:
+      - !type:MinSolution
+        solution: sauce
+        reagent:
+          ReagentId: Ketchup
+        quantity: 5
```
Comment (moonheart08) [inline Resources/Prototypes/Recipes/Construction/Graphs/food/pizza.yml:21]: ideally should allow any liquid sauce to be used here, perhaps future work

**Comment by Arctic-Husky** [comment]
I'm a little confused, will this allow for making pizza of any kind, or will it only follow recipes, or both? Could I make a pizza with just cheese for example?

**Comment by lzk228** [comment]
> I'm a little confused, will this allow for making pizza of any kind, or will it only follow recipes, or both? Could I make a pizza with just cheese for example? Sounds really cool!

it will be recipes like now, but instead of putting everything into a microwave, you should really make a pizza

**Comment by deltanedas** [comment]
no you can make custom pizza like 100 banana peels or 10 cheese (which would look like a margherita pizza but have a lot more reagents) it gets a name based on the most common ingredient
in the future maybe there will be a visualiser for each topping so you can see all the stuff you add

**Comment by UbaserB** [comment]
is it possible to make organ pizza

**Comment by deltanedas** [comment]
wyapttb
when you add PizzaTopping tag bozo :trollface:

**Comment by potato1234x** [comment]
give everything pizza topping tag :trollface:

**Comment by Boaz1111** [comment]
Based and chefpilled

**Comment by 2013HORSEMEATSCANDAL** [comment]
ive had an idea for implementing recipes for things other than pizzas, how do I make sure we aren't stepping on each others toes here ?

**Comment by deltanedas** [comment]
it can be used for non pizza things, just add a separate constructedMeal for salad/soup/whatever and use meal: Salad or whatever in each recipe

**Comment by deltanedas** [comment]
:warning: ok im gonna close it for now unless i get some inspiration for stuff or someone else revive

**Comment by Boaz1111** [comment]
> :warning: ok im gonna close it for now unless i get some inspiration for stuff or someone else revive

I'm gonna revive this PR, can you elaborate on what you mean by "some inspiration for stuff", and what some notable things are that I should keep in mind while working on it

**Comment by deltanedas** [comment]
look at the big todo list in pr desc

**Comment by deltanedas** [comment]
to anyone else who thinks they can revive it, it will need guidebook stuff too

**Comment by USMP-lancer** [comment]
the pizza pie power

**Pull request closed without merging**